app/src/main/java/me/guillaumin/android/osmtracker/gpx/TrackFileProvider.java

```diff
@@ -0,0 +1,121 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ */
+
+public final class TrackFileProvider extends ContentProvider {
+    private static final String TAG = ExportToTempFileTask.class.getSimpleName();
+
+    @Override
+    public boolean onCreate() {
+        return false;
+    }
+
+    @Override
+    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
+        return new AbstractCursor() {
+            @Override
+            public int getCount() {
+                return 0;
+            }
+
+            @Override
+            public String[] getColumnNames() {
+                return new String[0];
+            }
+
+            @Override
+            public String getString(int column) {
+                return null;
+            }
+
+            @Override
+            public short getShort(int column) {
+                return 0;
+            }
+
+            @Override
+            public int getInt(int column) {
+                return 0;
+            }
+
+            @Override
+            public long getLong(int column) {
+                return 0;
+            }
+
+            @Override
+            public float getFloat(int column) {
+                return 0;
+            }
+
+            @Override
+            public double getDouble(int column) {
+                return 0;
+            }
+
+            @Override
+            public boolean isNull(int column) {
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public String getType(Uri uri) {
```

Comment:
It also seems to affect the mail sharing option, when I try to open the ZIP in my mail client (Thunderbird) it opens it with a text editor rather than the archive tool. Looking at the mail source shows a type of `application/xml` for the ZIP attachment:

```
------NBUC58F9SIXAUMV2UF11EAMGBWUN0X
Content-Type: application/xml;
 name="2017-07-16_11-40-37.zip"
Content-Transfer-Encoding: base64
Content-Disposition: attachment;
 filename="2017-07-16_11-40-37.zip";
 size=-1

UEsDBBQACAgIACFd8EoAAAAAAAAAAAAAAAAXAAAAMjAxNy0wNy0xNl8xMS00MC0zNy5ncHi91ttu
2yAYAODr9imY723AxqfIcVVt2m5arVIzqVpVTcyhDosDEaaLH2BPskfbkwzsTq2d5tBK81UI/vn5
wBycnTWrCvxkquZSTB3sIQcwUcg5F+XU+TL76CYOOMtPs3LdABMq6qmz0Ho9gXCz2XharmWp6D1v
...
```